deploy/util/io.py

```diff
@@ -46,16 +46,18 @@ def downloadURI(uri: str, local_dest_path: str, retries: int = 4) -> None:
     lpath = Path(local_dest_path)
     if lpath.exists():
         rootLogger.debug(f"Overwriting {lpath.resolve(strict=False)}")
-    rootLogger.debug(f"Downloading '{uri}' to '{lpath}'")
 
-    for attempt in range(retries):
+    assert tries > 0, "tries argument must be larger than 0"
+    for attempt in range(tries):
+        rootLogger.debug(f"Download attempt {attempt+1} of {tries}: '{uri}' to '{lpath}'")
         try:
             fs, rpath = url_to_fs(uri)
```

Comment:
Are you sure you want the `url_to_fs()` call in the try block?  It seems like exceptions arising here may be indicative of a larger issue that shouldn't be retried?

Reply:
For background, these two runs: 
* https://github.com/firesim/firesim/actions/runs/4301469200/jobs/7498730505
* https://github.com/firesim/firesim/actions/runs/4300595364/jobs/7496935845

fail for two different reasons on the same line:
* `aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host firesim-ci-vitis-xclbins.s3.***.amazonaws.com:443 ssl:default [Name or service not known]`
* `aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host firesim-ci-vitis-xclbins.s3.***.amazonaws.com:443 ssl:default [Temporary failure in name resolution]`

I added url_to_fs() inside the block because I wasn't sure if it did the DNS resolution or not. I will move it out. Thanks for reviewing this